tests: JUnit4 + PageObjects + Selenium fixes; add Allure; green 12/12…

PavAnufriev opened the pull request:
Состав изменений

Page Object для ключевых страниц (главная, вход, регистрация, восстановление).
Набор UI‑тестов на JUnit 4 и Selenium 4.
WebDriverManager для управления драйверами.
Интеграция Allure, шаги и аттачи.
Генератор тестовых данных (email, пароли).
Покрытие

Регистрация: успешная, ошибка при пароле < 6 символов.
Вход: через “Войти в аккаунт”, “Личный кабинет”, из форм регистрации и восстановления пароля.
Конструктор: переходы по разделам “Булки”, “Соусы”, “Начинки”.
Окружение

JDK 11, Maven, JUnit 4.13.2, Selenium 4.15, Allure 2.
Браузеры: Google Chrome и Яндекс.Браузер.
Результаты

Локальный прогон прошёл успешно; отчёт Allure сформирован.

PavAnufriev committed:
tests: JUnit4 + PageObjects + Selenium fixes; add Allure; green 12/12; generate allure report

okryukova commented on .allure/allure-2.24.0/bin/allure:
⛔️Нужно исправить.  Папку .allure не нужно было загружать в репозиторий. Эта папка должна быть добавлена в .gitignore.

okryukova commented on src/test/java/ru/netology/stellarburgers/tests/ConstructorTest.java:
```diff
@@ -0,0 +1,83 @@
+public class ConstructorTest extends BaseTest {
+    
+    @Test
+    @Description("Проверка работы перехода к разделу 'Булки' в конструкторе")
```
⚠️Можно улучшить. В allure есть еще одна аннотация для описания теста DisplayName. Добавь ее, пожалуйста, во всех тестах проекта

okryukova commented on src/test/java/ru/netology/stellarburgers/tests/LoginTest.java:
```diff
@@ -0,0 +1,116 @@
+        testPassword = TestDataGenerator.generateValidPassword();
+        
+        // Регистрируем пользователя
+        MainPage mainPage = new MainPage(driver);
```
⛔️Нужно исправить. Согласно заданию, тестового пользователя нужно создать с помощью АПИ запроса. Это должен быть рандомный пользователь, уникальный для каждого теста. После выполнения теста его нужно удалить.  Поправь, пожалуйста, этот момент во всем коде

okryukova commented on src/test/java/ru/netology/stellarburgers/utils/TestDataGenerator.java:
```diff
@@ -0,0 +1,54 @@
+    /**
+     * Генерирует случайное имя
+     */
+    public static String generateRandomName() {
```
⚠️Можно улучшить. Генератор данных это отлично, но лучше использовать готовые решения JavaFaker/DataFaker, так как это сокращает нашу работу.

okryukova commented on src/test/java/ru/netology/stellarburgers/utils/TestDataGenerator.java:
```diff
@@ -0,0 +1,54 @@
+     */
+    public static String generateInvalidPassword() {
+        return generateRandomPassword(5);
+    }
```
⛔️Нужно исправить.  Нет отчета о тестировании

okryukova replied:
нет отчета

okryukova replied:
Не вижу отчет. папка allure-results.

PavAnufriev committed:
Рефакторинг тестов согласно ревью: добавлен JavaFaker для генерации данных, API-клиент для setup/teardown пользователей, улучшена отчетность и описания тестов

okryukova commented on .allure/allure-2.24.0/bin/allure:
⛔️Нужно исправить. Папку .allure не нужно было загружать в репозиторий. Эта папка должна быть добавлена в .gitignore.

PavAnufriev committed:
fix: исправлены замечания по качеству кода

- Удалены неиспользуемые импорты (User, By)
- Добавлены аннотации @Override для методов tearDown()
- Исправлена конфигурация junit-platform.properties для JUnit 4
- Все тесты проходят успешно (12/12)

PavAnufriev committed:
docs: добавлен полный отчет Allure о тестировании

- Добавлена папка allure-results/ с сырыми данными тестов
- Добавлена папка allure-report/ со статическим HTML отчетом
- Все 12 тестов прошли успешно (100% success rate)
- Покрытие: регистрация, вход, конструктор бургеров
- README с инструкциями по просмотру отчета
